src/Gui/CommandPalette.cpp

```diff
@@ -0,0 +1,443 @@
+
+    QColor textColor = option.palette.color(
+        isEnabled ? QPalette::Normal : QPalette::Disabled,
+        option.state & QStyle::State_Selected ? QPalette::HighlightedText : QPalette::Text
+    );
+    painter->setPen(textColor);
+
+    QRect titleRect = textRect;
+    if (!tooltip.isEmpty()) {
+        titleRect.setHeight(textRect.height() / 2);
+    }
+
+    // draw group name on the right if available
+    if (!groupName.isEmpty()) {
+        QFont groupFont = titleFont;
+        groupFont.setPointSize(qMax(groupFont.pointSize() - 1, 8));
+        painter->setFont(groupFont);
+
+        QColor groupColor = textColor;
+        // make group name transparent
+        groupColor.setAlpha(150);
+        painter->setPen(groupColor);
+
+        QFontMetrics groupFm(groupFont);
+        int groupWidth = groupFm.horizontalAdvance(groupName);
+
+        QRect groupRect = titleRect;
+        groupRect.setLeft(titleRect.right() - groupWidth);
+
+        painter->drawText(groupRect, Qt::AlignRight | Qt::AlignVCenter, groupName);
+
+        // adjust title rect to not overlap with group name
+        titleRect.setRight(groupRect.left() - 10);
+
+        // reset font and color for title
+        painter->setFont(titleFont);
+        painter->setPen(textColor);
+    }
+
+    painter->drawText(titleRect, Qt::AlignLeft | Qt::AlignVCenter, title);
+
+    // draw tooltip/description and description if it is avaialble
+    if (!tooltip.isEmpty()) {
+        QFont tooltipFont = option.font;
+        tooltipFont.setPointSize(qMax(tooltipFont.pointSize() - 1, 8));
+        painter->setFont(tooltipFont);
+
+        QColor tooltipColor = textColor;
+        // make the tooltip a bit transparent
+        tooltipColor.setAlpha(180);
+        painter->setPen(tooltipColor);
+
+        QRect tooltipRect = textRect;
+        tooltipRect.setTop(titleRect.bottom());
+
+        // strip HTML tags from tooltip
+        QTextDocument doc;
+        doc.setHtml(tooltip);
+        QString plainText = doc.toPlainText();
+
+        // elide if desc is too long
+        QFontMetrics fm(tooltipFont);
+        QString elidedTooltip = fm.elidedText(plainText, Qt::ElideRight, tooltipRect.width());
+
+        painter->drawText(tooltipRect, Qt::AlignLeft | Qt::AlignVCenter, elidedTooltip);
+    }
+
+    painter->restore();
+}
+
+QSize CommandItemDelegate::sizeHint(const QStyleOptionViewItem& option, const QModelIndex& index) const
+{
+    const QString tooltip = index.data(Qt::ToolTipRole).toString();
+
+    // if there's a tooltip, make the item taller
+    const double height = option.fontMetrics.height() * (tooltip.isEmpty() ? 1.5 : 2.8);
+
+    return {option.rect.width(), static_cast<int>(height)};
+}
+
+////////////////////// CommandPalette implementation //////////////////////
+
+CommandPalette::CommandPalette(QWidget* parent)
+    : QDialog(parent)
+{
+    setupUi();
+
+    setModal(false);
+    setWindowFlags(Qt::Dialog | Qt::FramelessWindowHint);
```

Comment:
```suggestion
    setWindowFlags(Qt::Dialog | Qt::FramelessWindowHint | Qt::BypassWindowManagerHint);
```

Without the bypass window manager hint it still can show frames sometimes